internal/js_library/js_library.bzl

```diff
@@ -50,95 +56,147 @@ def write_amd_names_shim(actions, amd_names_shim, targets):
     actions.write(amd_names_shim, amd_names_shim_content)
 
 def _impl(ctx):
-    files = []
+    files = ctx.files.srcs + ctx.files.named_module_srcs
     typings = []
     js_files = []
-
-    for src in ctx.files.srcs:
-        if src.is_source and not src.path.startswith("external/"):
-            dst = ctx.actions.declare_file(src.basename, sibling = src)
-            if ctx.attr.is_windows:
-                copy_cmd(ctx, src, dst)
-            else:
-                copy_bash(ctx, src, dst)
-            if dst.basename.endswith(".d.ts"):
-                typings.append(dst)
-            else:
-                files.append(dst)
-        elif src.basename.endswith(".d.ts"):
-            typings.append(src)
-        else:
-            files.append(src)
-
-    for p in files:
-        if p.basename.endswith(".js") or p.basename.endswith(".js.map") or p.basename.endswith(".json"):
-            js_files.append(p)
+    include_npm_package_info = False
+
+    for file in files:
+        if file.basename.endswith(".js") or file.basename.endswith(".js.map") or file.basename.endswith(".json"):
```

Comment:
we always forget .jsx

Reply:
@alexeagle I look at `.jsx` (and also `.tsx`) as source files that should be first processed before be able to ship in a node module and therefore probably should not be included in the js_library too. What are your thoughts around it?

Reply:
I'm not sure. I guess if packages on npm rarely include jsx then you're right, should be only internal within a package

Reply:
yeah that is case. We can revisit later if we found the other way around!